A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_AirwaysFromWaypointPage.js

```diff
@@ -45,79 +54,97 @@ class A320_Neo_CDU_AirwaysFromWaypointPage {
+                                const lastWaypoint = mcdu.flightPlanManager.getWaypoints()[mcdu.flightPlanManager.getEnRouteWaypointsLastIndex()];
+                                if (lastWaypoint.infos instanceof IntersectionInfo || lastWaypoint.infos instanceof VORInfo || lastWaypoint.infos instanceof NDBInfo) {
+                                    const airway = lastWaypoint.infos.airways.find(a => {
+                                        return a.name === value;
+                                    });
+                                    if (airway) {
+                                        A320_Neo_CDU_AirwaysFromWaypointPage.ShowPage(mcdu, waypoint, offset, airway);
+                                    } else {
+                                        mcdu.showErrorMessage("AWY/WPT MISMATCH");
+                                    }
                                 }
-                            }
+                            });
                         }
                     };
-                } else {
-                    rows[i] = [pendingAirway.name, "[ ][color]blue"];
+                } else if (pendingAirway) {
+                    subRows[i] = ["VIA", "TO"];
+                    rows[i] = [`${pendingAirway.name}[color]blue`, "[ ][color]blue"];
                     mcdu.onRightInput[i] = () => {
                         const value = mcdu.inOut;
                         if (value.length > 0) {
                             mcdu.clearUserInput();
-                            mcdu.insertWaypointsAlongAirway(value, mcdu.flightPlanManager.getEnRouteWaypointsLastIndex() + 1, pendingAirway.name, (result) => {
-                                if (result) {
-                                    A320_Neo_CDU_AirwaysFromWaypointPage.ShowPage(mcdu, waypoint, offset);
-                                } else {
-                                    mcdu.showErrorMessage("AWY/WPT MISMATCH");
-                                }
+                            mcdu.ensureCurrentFlightPlanIsTemporary(() => {
+                                mcdu.insertWaypointsAlongAirway(value, mcdu.flightPlanManager.getEnRouteWaypointsLastIndex() + 1, pendingAirway.name, (result) => {
+                                    if (result) {
+                                        A320_Neo_CDU_AirwaysFromWaypointPage.ShowPage(mcdu, waypoint, offset);
+                                    } else {
+                                        mcdu.showErrorMessage("AWY/WPT MISMATCH");
+                                    }
+                                });
                             });
                         }
                     };
-                    if (rows[i + 1]) {
-                        rows[i + 1] = ["-----"];
-                    }
+                    // May be useful for intersecions.
+                    // if (rows[i + 1]) {
+                    //     rows[i + 1] = ["-----"];
+                    // }
```

Comment:
Please remove commented code throughout your changes. Thanks !

Reply:
Forgot to remove, thanks.